src/CombinedSubscription.js

```diff
@@ -0,0 +1,74 @@
+import HistoricalSubscription from './HistoricalSubscription'
+import RealTimeSubscription from './RealTimeSubscription'
+import AbstractSubscription from './AbstractSubscription'
+
+export default class CombinedSubscription extends AbstractSubscription {
+    constructor(
+        streamId, streamPartition, callback, options, groupKeys,
+        propagationTimeout, resendTimeout
+    ) {
+        super(streamId, streamPartition, callback, groupKeys, propagationTimeout, resendTimeout, true)
+        this.sub = new HistoricalSubscription(streamId, streamPartition, callback, options, groupKeys, this.propagationTimeout, this.resendTimeout)
+        this.sub.on('resend done', async (lastReceivedMsgRef) => {
+            const realTime = new RealTimeSubscription(
+                streamId, streamPartition, callback, groupKeys, this.propagationTimeout, this.resendTimeout, lastReceivedMsgRef
+            )
+            await Promise.all(this.sub.realTimeMsgsQueue.map((msg) => realTime.handleBroadcastMessage(msg, () => true)))
+            this.sub.stop()
+            this.sub = realTime
+        })
+        this.sub.on('done', () => this.emit('done'))
```

Comment:
Should any of these event listeners be bound to  `realTime` when the switch from historical to realtime happens?